src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy

```diff
@@ -46,5 +46,20 @@ class GitVersionPlugin implements Plugin<Project> {
             println project.version
         }
     }
-}
 
+    private File findRootGitDir(File currentRoot) {
+        File gitDir = new File(currentRoot, '.git')
+
+        if (gitDir.exists()) {
+            return gitDir
+        }
+
+        // stop at the root directory, return non-existing File object
+        if (currentRoot.parentFile == null) {
```

Comment:
1. switched to using `File` methods
2. detecting root by using `File::parentFile == null`

this should be safe on windows. don't have a win machine to test on though :-\